llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -3260,26 +3260,10 @@ static SDValue extractBooleanFlip(SDValue V, SelectionDAG &DAG,
   if (V.getOpcode() != ISD::XOR)
     return SDValue();
 
-  ConstantSDNode *Const = isConstOrConstSplat(V.getOperand(1), false);
-  if (!Const)
+  if (!isConstOrConstSplat(V.getOperand(1), false))
```

Comment:
Move this after the isBoolConstant() check (which does something similar)? It looks like you only need it for the `if (Force)` case now.

Reply:
For what it's worth there's no test difference with this check removed, but I stuck it back in anyway just to avoid changing the Force == true codepath